### PR TITLE
Fix PED bugs and add edit command Uml

### LINKED ISSUE #313: [PE-D][Tester A] Incorrect user guide instruction - createGroup, addToGroup, removeFromGroup
**Description**
The field 'INDICES' in createGroup, addToGroup, removeFromGroup instructions in user guide are in [square bracket], which is indicated as optional in the note. However, 'INDICES' seems to be mandatory to be executed.

**Expected**
Able to execute 
`createGroup g/blood drive 2024`, 

**Actual**
Error message displayed

**Screenshots**
*Figure 1: Instructions on user guide*

*Figure 2: Error message*

**Note**
Similar issues occurred in addToGroup, removeFromGroup.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: LynetteLeeSiYing/ped#1

### LINKED ISSUE #325: [PE-D][Tester E] User Guide: Target user not specified clearly
At the start of the User Guide, the target user of the app is not specified clearly. The description is quite general and perhaps could have more elaboration on the target user and value proposition to the target user. 

These were clearly stated in the DG but I believe they need to be stated in the UG as well.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: tingxuanp/ped#4

### LINKED ISSUE #329: [PE-D][Tester A] Commands are case-sensitive
**Description**
Commands are case-sensitive which is inconvenient to user

**Steps to reproduce**
execute any command with some uppercase letters

**Expected**
works as usual

**Actual**
Showing error message as unknown command

**Screenshots**

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: LynetteLeeSiYing/ped#6

### LINKED ISSUE #330: [PE-D][Tester A] Same name with spaces are treated as different person
**Description**
Same name with spaces are treated as different person

**Steps to reproduce**
add another person with same name as existing person but with extra spaces in between

**Expected**
Warnings to user as duplicate person

**Actual**
Added successfully

**Screenshots**

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: LynetteLeeSiYing/ped#7

### LINKED ISSUE #342: [PE-D][Tester A] Same group name treated differently
**Description**
Same group name with spaces are treated differently. 

**Steps to reproduce**
Create a group which has the same name as existing group but with extra spaces in between.

**Expected**
Warning shown as duplicate group names.

**Actual**
Group created successfully

**Screenshots**

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: LynetteLeeSiYing/ped#12

### LINKED ISSUE #345: [PE-D][Tester B] App allows users to add in emails without a period symbol in the domain
**Description**: The AddressBook app allows users to add emails that do not have the period symbol, potentially recording erroneous emails.

**Steps to Reproduce**:

1. Open the AddressBook app.
2. Enter a command to add a new contact with an email without a '.' in the domain.

   Example command:
   ```plaintext
   add r/volunteer n/John Doe p/98765432 e/johnd@examplecom a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney h/30

**Expected Result**:
The app rejects the contact information with an error message.

**Actual Result**:
The app accepts the contact information.


![image.png](https://raw.githubusercontent.com/Runus01/ped/main/files/bc4fe7fc-b645-41e3-884c-9428e56e4466.png)



<!--session: 1731055275750-295d3d6f-a0c8-4807-9fca-76c1c5787955-->
<!--Version: Desktop v3.5.4-->

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: Runus01/ped#6

### LINKED ISSUE #346: [PE-D][Tester D] Bugs in email format check
As shown in the picture above, the email is obviously invalid in format, but the system still perform the edit operation. It will be better to enforce the email format check in the address book.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: jhqiu21/ped#7

### LINKED ISSUE #353: [PE-D][Tester D] The system do not allow to create an empty group
As shown in the picture above, trying to create a group without person now will lead to invalid command. However in real life, it will be much common for person create group first before the activity, and the owner try to add the relevant person later when the owner get the person list. Hence, it will be better to make member field optional. In other word, a group can persons or not.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: jhqiu21/ped#8

### PR DESCRIPTION
**Pull request by chongsun2002**
Fixes #313 (Edited command summary)
Fixes #325 (Edited start of UG)
Fixes #329 (Edited planned features)
Fixes #342 (Stated in createGroup)
Fixes #345 , Fixes #346  (Just stated that some emails have no periods in the UG) and fixes #330.
image
Fixes #353 (Not this specific PR, but has been fixed)
Partial F1x for #377

### CODE REVIEW

**Review by seah-minlong — APPROVED** [review]
LGTM

**Review by seah-minlong — CHANGES_REQUESTED** [review]
Failing CI

**Comment by chongsun2002** [comment]
No newline at EOF is a crazy way to fail CI

File: docs/UserGuide.md [hunk 1]
```diff
@@ -66,7 +67,10 @@ ContactsForGood (CFG) is a **desktop app for managing contacts, optimized for us
       will result in an error)
 
 * Items in square brackets are optional.<br>
-  e.g `n/NAME [t/TAG]` can be used as `n/John Doe t/friend` or as `n/John Doe`.
+  e.g `n/NAME [t/TAG]` can be used as `n/John Doe t/friend` or as `n/John Doe`.  
+  Note: If a field is optional and you choose to not input anything into the field, **DO NOT**
+  leave an empty prefix, as this will not be a valid command.
```
Comment (seah-minlong) [inline docs/UserGuide.md:72]: Sounds abit fierce. Should we change it

**Review by seah-minlong — APPROVED** [review]
LGTM

**Pull request merged**